### PR TITLE
Mi-go furnitures contain a fetid goop that can mask your scent

### PR DESCRIPTION
**Pull request by Fris0uman**
#### Summary
```SUMMARY: Content "Add fetid goop that can change your scent"```

#### Purpose of change
Add infrastructure for item to change the player scent
Add scents_ignored entry to mtype, list of scent ignored by a monster type.
Add fetid goop to migo location that can be used to get a scent that Mi-gos ignore
       The effects last one hour, the scent can be washed away by water, using the goop makes you sick

#### Describe the solution
Implement change_scent_iuse
Add appropriate checks to revert scent to normal after a time or have the scent washed away by water
Implement scents_ignored
#### Testing
Use the goop
debug scent map to check the scent has change
Check that Migos don't follow the scent
Check that the palyer does get sick
Jump in water > scent returns to normal

#### Additional context
The criteria for water removing the scent is having 50% of max weetness on torso, I did not manage to get that from spending a short time under the rain, but jumping in deep water does the trick.
Sadly washing away the smell with water does not remove the side effect, I don't know how to make that work.

### CODE REVIEW

**Comment by kevingranade** [comment]
Rather than obliterating scent, I'd much prefer that any kind of masking reduces effective strength of scent.  This is also complicated in that why would everything ignore the mi-go goop smell?  If anything this makes your smell stronger, and as long as you aren't in a mi-go lair it's just as distinctive as smelling like a person.

**Comment by Fris0uman** [comment]
So yes, but also why would any creature follow that horrid smell?

File: src/iuse_actor.cpp [hunk 1]
```diff
@@ -4709,3 +4709,47 @@ std::unique_ptr<iuse_actor> sew_advanced_actor::clone() const
...
 }
+
+void change_scent_iuse::load( const JsonObject &obj )
+{
+    scenttypeid = scenttype_id( obj.get_string( "scent_typeid" ) );
+    if( !scenttypeid.is_valid() ) {
+        obj.throw_error( "Invalid scent type id.", "scent_typeid" );
+    }
+    if( obj.has_array( "effects" ) ) {
+        JsonArray jsarr = obj.get_array( "effects" );
```
Comment (BevapDin) [inline src/iuse_actor.cpp]:
```suggestion
        for( JsonObject e : obj.get_array( "effects" ) ) {
```
This line replaces three lines in your code.

File: src/iuse_actor.cpp [hunk 2]
```diff
@@ -4709,3 +4709,47 @@ std::unique_ptr<iuse_actor> sew_advanced_actor::clone() const
...
+int change_scent_iuse::use( player &p, item &it, bool, const tripoint & ) const
+{
+    p.set_value( "prev_scent", p.get_type_of_scent().c_str() );
+    if( waterproof ) {
+        p.set_value( "waterproof_scent", "true" );
+    }
+    p.add_effect( efftype_id( "masked_scent" ), duration, num_bp, false, scent_mod );
+    p.set_type_of_scent( scenttypeid );
+    p.mod_moves( -moves );
+    add_msg( m_info, string_format( _( "You use the %s to mask your scent" ), it.tname() ) );
```
Comment (BevapDin) [inline src/iuse_actor.cpp]:
```suggestion
    add_msg( m_info, _( "You use the %s to mask your scent" ), it.tname() );
```

File: src/iuse_actor.cpp [hunk 3]
```diff
@@ -4709,3 +4709,47 @@ std::unique_ptr<iuse_actor> sew_advanced_actor::clone() const
...
+
+    // Apply the various effects.
+    for( const auto &eff : effects ) {
+        time_duration dur = eff.duration;
+        p.add_effect( eff.id, dur, eff.bp, eff.permanent );
```
Comment (BevapDin) [inline src/iuse_actor.cpp]:
```suggestion
        p.add_effect( eff.id, eff.duration, eff.bp, eff.permanent );
```
There is no need to copy this into a local variable first.

**Pull request merged**